modules/gdscript/gdscript_analyzer.cpp

```diff
@@ -313,6 +313,136 @@ Error GDScriptAnalyzer::resolve_inheritance(GDScriptParser::ClassNode *p_class,
 
+	#define CHECK_MEMBER_AGAINST_INHERITANCE(name, node) \
+	{ \
+		if (current_class_node->members_indices.has(name)) { \
+			int index = current_class_node->members_indices[name]; \
+			const GDScriptParser::ClassNode::Member *base_member = &current_class_node->members[index]; \
+			\
+			if (base_member->type == GDScriptParser::ClassNode::Member::VARIABLE || \
+				base_member->type == GDScriptParser::ClassNode::Member::CONSTANT || \
+				base_member->type == GDScriptParser::ClassNode::Member::ENUM || \
+				base_member->type == GDScriptParser::ClassNode::Member::ENUM_VALUE || \
+				base_member->type == GDScriptParser::ClassNode::Member::CLASS) { \
+					push_error(vformat(R"(The member "%s" already exists in a parent class.)", name), node); \
+					return ERR_PARSE_ERROR; \
+			} else if (base_member->type == GDScriptParser::ClassNode::Member::SIGNAL && p_class->members[i].type == GDScriptParser::ClassNode::Member::SIGNAL) { \
+				push_error(vformat(R"(The signal "%s" already exists in a parent class.)", name), node); \
+				return ERR_PARSE_ERROR; \
+			} \
+		} \
+	}
+
+	// Check for invalid naming in members
+	GDScriptParser::DataType *current_data_type = &p_class->base_type;
+	while (current_data_type && current_data_type->kind == GDScriptParser::DataType::Kind::CLASS) {
+		GDScriptParser::ClassNode *current_class_node = current_data_type->class_type;
+		for (int i = 0; i < p_class->members.size(); i++) {
+			switch(p_class->members[i].type) {
+				case GDScriptParser::ClassNode::Member::VARIABLE:
+					CHECK_MEMBER_AGAINST_INHERITANCE(p_class->members[i].variable->identifier->name, p_class->members[i].variable)
+					break;
+				case GDScriptParser::ClassNode::Member::CONSTANT:
+					CHECK_MEMBER_AGAINST_INHERITANCE(p_class->members[i].constant->identifier->name, p_class->members[i].constant)
+					break;
+				case GDScriptParser::ClassNode::Member::ENUM:
+					CHECK_MEMBER_AGAINST_INHERITANCE(p_class->members[i].m_enum->identifier->name, p_class->members[i].m_enum)
+					break;
+				case GDScriptParser::ClassNode::Member::ENUM_VALUE:
+					CHECK_MEMBER_AGAINST_INHERITANCE(p_class->members[i].enum_value.identifier->name, p_class->members[i].enum_value.custom_value)
+					break;
+				case GDScriptParser::ClassNode::Member::CLASS:
+					CHECK_MEMBER_AGAINST_INHERITANCE(p_class->members[i].m_class->identifier->name, p_class->members[i].m_class)
+					break;
+				case GDScriptParser::ClassNode::Member::SIGNAL:
+					CHECK_MEMBER_AGAINST_INHERITANCE(p_class->members[i].signal->identifier->name, p_class->members[i].signal)
+					break;
+				default:
+					break;
+			}
+		}
+		current_data_type = &current_class_node->base_type;
+	}
+
+	#define CHECK_MEMBER_AGAINST_NATIVE(name, native_type_string, node, is_signal) \
+	if (is_signal) { \
+		if (ClassDB::has_signal(native_type_string, name)) { \
+			push_error("Member '" + name + "' redefined (original in native class '" + String(native_type_string) + "')", node); \
```

Comment:
You should use the `vformat` style and use double quotes (`"`) to surround names so it follows the standard (using raw strings makes it easier).